cl/users/email_handlers.py

```diff
@@ -417,7 +403,41 @@ def is_message_stored(message_id: str) -> tuple[bool, int | None]:
     return False, None
 
 
-from cl.users.tasks import send_failed_email
+def schedule_failed_email(recipient_email: str) -> None:
+    """Schedule recipient's waiting failed emails after receiving a delivery
+    notification because it means the recipient's inbox is working again.
+
+    :param recipient_email: The recipient's email address to whom the delivery
+    notification belongs
+    :return: None
+    """
+
+    # Look for FailedEmail objects with WAITING status.
+    failed_messages = FailedEmail.objects.filter(
+        recipient=recipient_email, status=STATUS_TYPES.WAITING
+    ).order_by("date_created")
+    # Before clean possible failed task objects.
+    delete_failed_tasks_objects(recipient_email)
+
+    # Get the next retry datetime to schedule the message, in this case now()
+    scheduled_datetime = get_next_retry_date(recipient_email)
+    # If the recipient has multiple messages to retry we'll send slowly, one
+    # message per minute.
+    minutes = 1
+    for fail_message in failed_messages:
+        # Set schedule time and update status to ENQUEUED_DELIVERY
+        next_retry_datetime = scheduled_datetime + timedelta(minutes=minutes)
+        fail_message.next_retry_date = next_retry_datetime
+        fail_message.status = STATUS_TYPES.ENQUEUED_DELIVERY
+        fail_message.save()
+        # Call celery task.
+        send_failed_email.si(fail_message.pk).apply_async(
+            eta=next_retry_datetime
+        )
+        minutes += 1
+
+
+from cl.users.tasks import check_recipient_deliverability, send_failed_email
```

Comment:
Why are we doing these imports on line ~450 instead of at the top? We don't usually do that, except when we need to fix a circular import propblem.

Reply:
Yeah, there's a circular import problem here, since the method below`enqueue_email` is also imported in `cl.users.tasks` so I was thinking about solving this issue by putting it directly into `cl.users.tasks`, I was not sure what would you prefer, let me know if I should move this.